common/djangoapps/student/management/commands/bulk_unenroll.py

```diff
@@ -27,39 +27,75 @@ def add_arguments(self, parser):
         parser.add_argument('-p', '--csv_path',
                             metavar='csv_path',
                             dest='csv_path',
-                            required=True,
+                            required=False,
                             help='Path to CSV file.')
 
     def handle(self, *args, **options):
+
         csv_path = options['csv_path']
-        with open(csv_path) as csvfile:
-            reader = unicodecsv.DictReader(csvfile)
-            for row in reader:
-                username = row['username']
-                email = row['email']
-                course_key = row['course_id']
-                try:
-                    user = User.objects.get(Q(username=username) | Q(email=email))
-                except ObjectDoesNotExist:
-                    user = None
-                    msg = 'User with username {} or email {} does not exist'.format(username, email)
-                    logger.warning(msg)
+        if csv_path:
+            with open(csv_path) as csv_file:
+                self.un_enroll_users(csv_file)
+        else:
+            csv_file = BulkUnenrollConfiguration.current().csv_file
+            self.un_enroll_users(csv_file)
+
+    def un_enroll_users(self, csv_file):
+        reader = list(unicodecsv.DictReader(csv_file))
+        users = self.get_users_info(reader)
+        enrollments, course_ids = self.get_enrollments(reader, users)
+        users_unenrolled = []
+        for row in reader:
+            username = row['username']
+            email = row['email']
+            course_key = row['course_id']
+            try:
+                user = users.get(Q(username=username) | Q(email=email))
+            except ObjectDoesNotExist:
+                user = None
+                msg = 'User with username {} or email {} does not exist'.format(username, email)
+                logger.warning(msg)
 
+            try:
+                course_id = course_ids[course_key]
+            except KeyError:
+                course_id = None
+                msg = 'Invalid course id {course_id}, skipping un-enrollement for {username}, {email}'.format(**row)
+                logger.warning(msg)
+
+            try:
+                enrollments.get(user_id=user.id, course_id=course_id)
```

Comment:
Remove this, this is unnecessary. I believe unenroll will also raise same exception if enrollment doesn't exist.